### ISSUE TITLE
[A11y] Semantics roles system ☂️

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#159741]
### Use case

Our current semantics system is missing a bunch of roles in web and desktop platform, and some in mobile platform.
See
http://flutter.dev/go/semantics-roles

### Proposal

http://flutter.dev/go/semantics-role-system

### General
- [x] https://github.com/flutter/flutter/issues/161346

### Dropdown, Menu, ComboBox
- [x] https://github.com/flutter/flutter/issues/157177

### Tabs
- [x] https://github.com/flutter/flutter/issues/107861 

### Table
- [x] https://github.com/flutter/flutter/issues/45205

### List
- [x] https://github.com/flutter/flutter/issues/144988

### menu
- [ ] https://github.com/flutter/flutter/issues/161561

##### Link
- [x] https://github.com/flutter/flutter/issues/157210
- [x] https://github.com/flutter/flutter/issues/102535

##### Dialog
- [x] https://github.com/flutter/flutter/issues/157204
- [x] https://github.com/flutter/flutter/issues/157207
- [x] https://github.com/flutter/flutter/issues/162124

##### Heading
- [ ] https://github.com/flutter/flutter/issues/155928

##### form
- [ ] https://github.com/flutter/flutter/issues/161628

##### tooltip
- [ ] https://github.com/flutter/flutter/issues/161630

### misc
- [ ] https://github.com/flutter/flutter/issues/161539
- [ ] https://github.com/flutter/flutter/issues/161540
- [x] https://github.com/flutter/flutter/issues/161542
- [ ] https://github.com/flutter/flutter/issues/161557
- [ ] https://github.com/flutter/flutter/issues/161558
- [ ] https://github.com/flutter/flutter/issues/161559
- [ ] https://github.com/flutter/flutter/issues/161631
- [ ] https://github.com/flutter/flutter/issues/161689
- [ ] https://github.com/flutter/flutter/issues/161690

**Comment by chunhtai** [comment]
I am going to close this one since high priority ones were added, and will use the individual issue  to track remanding work
